File: earthlings/stepCounter.py
n=int(input())
a=list(map(int,input().strip().split()))[:n]
b=list(map(int,input().strip().split()))[:n]

#print(a, b)

count=0
minVal=min(a)
for i in range(0, n):
    while a[i]>minVal:
        a[i]=a[i]-b[i]
        count+=1
    if a[i]<minVal:
        minVal=a[i]
        i=0
    elif a[i]<0:
        count=-1
        break
    else:
        i+=1
            
print(count)            
'''
Problem

You are given two arrays
and . In each step, you can set  if . Determine the minimum number of steps that are required to make all

's equal.

Input format

    First line: 

 
Second line:
Third line:

Output format

Print the minimum number of steps that are required to make all

's equal. If it is not possible, then print -1.

Constraints

Sample input

2
5 6
4 3

Sample output

-1
Sample Input

5
5 7 10 5 15
2 2 1 3 5

Sample Output

8

Time Limit: 1
Memory Limit: 256
Source Limit: 
'''

File: earthlings/stepCounter.cpp
#include <iostream>

using namespace std;

int main(){

    int n;

cin>>n;

     int a[n], b[n], c=5001,d=0,e=0,f=0;

         for(int i=0; i<n; i++)

         { cin>>a[i]; if(a[i]<c) c=a[i]; }    

       for(int i=0; i<n; i++) cin>>b[i];

        

        for (int i = 1; i; i++) // INFINITE LOOP

       {

         for(int k=0; k<n;k++)

         {

               if(a[k]>c) {

                              if(a[k]>=b[k]) { a[k]= a[k] - b[k]; d+=1;}

                            }

               if(c>a[k]) c=a[k];  // MINIMUM VALUE IS UPDATED EACH TIME

         }

        

   for(int j=0; j<n;j++)

        { 

             if(a[j]!=c)

             {    if( a[j]<b[j] )

                 { cout<<-1; return 0 ;}

             }

         }

            

      for(int j=0; j<n-1;j++)

         {

             if(a[j]==a[j+1]) f=1;

             else {f=0; break;}

          }

 

            if(f==1) {cout<<d; return 0;}

        }

}